ngraph/core/reference/include/ngraph/runtime/reference/random_uniform.hpp

```diff
@@ -0,0 +1,263 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ctime>
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            std::pair<uint32_t, uint32_t> split_high_low(uint64_t value)
+            {
+                uint32_t low = static_cast<uint32_t>(value);
+                uint32_t high = static_cast<uint32_t>(value >> 32);
+                return {low, high};
+            }
+
+            uint64_t unite_high_low(uint32_t high, uint32_t low)
+            {
+                return (static_cast<uint64_t>(high) << 32) + low;
+            }
+
+            void calculate_round(const uint64_t& key, uint64_t& counter, uint64_t& n)
```

Comment:
Do we really need to have const reference to int?

Reply:
Applied this comment in separate PR: https://github.com/openvinotoolkit/openvino/pull/7012